lib/base/tlsutility.cpp

```diff
@@ -100,7 +100,9 @@ static void InitSslContext(const Shared<boost::asio::ssl::context>::Ptr& context
 		}
 	});
 #else /* OPENSSL_VERSION_NUMBER < 0x10100000L */
+#	ifdef SSL_OP_NO_RENEGOTIATION
 	flags |= SSL_OP_NO_RENEGOTIATION;
+#	endif /* SSL_OP_NO_RENEGOTIATION */
```

Comment:
Is there some macro that could be used to detect LibreSSL? If OpenSSL would change something with `SSL_OP_NO_RENEGOTIATION`  (something like rename it in v4, I hope they don't, but they probably could), this could would probably still compile but I'd argue that this would be a situation someone should look into, so it would be better if we check for the library we're writing this for rather than the existence of some constant.

Reply:
There should be both `LIBRESSL_VERSION_NUMBER` and `LIBRESSL_VERSION_TEXT` as described in [OPENSSL_VERSION_NUMBER(3)](https://man.egoslike.us/OPENSSL_VERSION_NUMBER.3). _(Note: man.openbsd.org has currently [operational difficulties](https://marc.info/?l=openbsd-misc&m=170335911200538&w=2), that's why this man mirror was used)_

Reply:
Normally we'd suggest *not* checking based on library name/version.

The same concern would apply to pretty much any other change that openssl could conceivably make to their API...